cmd/kubeadm/app/util/staticpod/utils_linux.go

```diff
@@ -156,3 +164,28 @@ func runKubeSchedulerAsNonRoot(pod *v1.Pod, runAsUser, runAsGroup *int64, update
 	pod.Spec.SecurityContext.RunAsGroup = runAsGroup
 	return nil
 }
+
+// runEtcdAsNonRoot updates the pod manifest and the hostVolume permissions to run etcd as non root.
+func runEtcdAsNonRoot(pod *v1.Pod, runAsUser, runAsGroup *int64, updatePathOwnerAndPermissions pathOwerAndPermissionsUpdaterFunc, cfg *kubeadmapi.ClusterConfiguration) error {
+	if err := updatePathOwnerAndPermissions(cfg.Etcd.Local.DataDir, *runAsUser, *runAsGroup, 0700); err != nil {
```

Comment:
this is an interesting one as during upgrade it would take the etcd data dir previously owned likely by root and recursively change all the etcd member files to be owned by a new user.

i'm interested if someone will complain about that WRT to users storing etcd on external drives...etc.

Reply:
re: 0700 why are we setting the executable bit BTW?

Reply:
/hold for this one Q

Reply:
Great question, we shouldn't be giving 0700 to files, they should be 0600. I had to do some refactoring, so please take another look.